inc/networkequipment.class.php

```diff
@@ -90,24 +92,26 @@ static function displayTabContentForItem(CommonGLPI $item, $tabnum=1, $withtempl
       global $CFG_GLPI;
 
       if ($item->getID() > 0) {
-         $pfNetworkEquipment = new PluginFusioninventoryNetworkEquipment();
+         $pfNetworkEquipment = new self();
+         $options = [
+            'target' => $CFG_GLPI['root_doc'].
+               '/plugins/fusioninventory/front/switch_info.form.php'
+         ];
+
+         switch (Session::getActiveTab('NetworkEquipment')) {
+            case 'PluginFusioninventoryNetworkEquipment$1':
+               $pfNetworkEquipment->showForm($item, $options);
+               break;
 
-         if (isset($_GET['displaysnmpinfo'])) {
-            $pfNetworkEquipment->showNetworkEquipmentInformation($item,
-                                                                 array('target'=>$CFG_GLPI['root_doc'].
-                                                                       '/plugins/fusioninventory/front/switch_info.form.php'));
-         } else {
-            $pfNetworkEquipment->showForm($item,
-                 array('target'=>$CFG_GLPI['root_doc'].
-                                    '/plugins/fusioninventory/front/switch_info.form.php'));
+            case 'NetworkPort$1':
+               $pfNetworkEquipment->showExtendedInfos($item, $options);
+               break;
          }
-         return TRUE;
       }
-      return FALSE;
+      return false;
```

Comment:
Method will always return `false`; is that intended?

Reply:
no it should return true, as for other objects